dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/GroupRestRepository.java

```diff
@@ -33,11 +40,46 @@
 
 @Component(GroupRest.CATEGORY + "." + GroupRest.NAME)
 public class GroupRestRepository extends DSpaceRestRepository<GroupRest, UUID> {
-    GroupService gs = EPersonServiceFactory.getInstance().getGroupService();
+    @Autowired
+    GroupService gs;
 
     @Autowired
     GroupConverter converter;
 
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @Override
+    protected GroupRest createAndReturn(Context context)
+            throws AuthorizeException, RepositoryMethodNotImplementedException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        GroupRest groupRest = null;
+
+        try {
+            groupRest = mapper.readValue(req.getInputStream(), GroupRest.class);
+        } catch (IOException excIO) {
+            throw new UnprocessableEntityException("error parsing the body... maybe this is not the right error code");
```

Comment:
This error message is a bit odd.  Perhaps we just say `"error parsing the body: " + excIO.getMessage()`?